partner_compassion/models/partner_compassion.py

```diff
@@ -528,18 +528,21 @@ def forget_me(self):
                 "geo_point": False,
                 "partner_latitude": False,
                 "partner_longitude": False,
+                "birthdate_date": False,
             }
         )
-        self.advocate_details_id.unlink()
-        self.survey_inputs.unlink()
-        self.env["mail.tracking.email"].search([("partner_id", "=", self.id)]).unlink()
-        self.env["auditlog.log"].search(
+        self._cr.execute("""update res_partner set ref='', global_id='' where id=""" + str(self.id))
+        self.advocate_details_id.sudo().unlink()
+        self.survey_inputs.sudo().unlink()
+        self.env["mail.tracking.email"].sudo().search([("partner_id", "=", self.id)]).unlink()
+        self.env["auditlog.log"].sudo().search(
             [("model_id.model", "=", "res.partner"), ("res_id", "=", self.id)]
         ).unlink()
-        self.env["partner.communication.job"].search(
+        self.env["partner.communication.job"].sudo().search(
             [("partner_id", "=", self.id)]
         ).unlink()
-        self.message_ids.unlink()
+        self.message_ids.sudo().unlink()
+        self.category_id.sudo().unlink()
```

Comment:
This is wrong, you will delete the tag instead of deleting the relation! I propose to add this above instead.
```suggestion
```